src/genophenocorr/model/_protein.py

```diff
@@ -141,6 +142,9 @@ def feature_type(self) -> FeatureType:
         """
         return self._type
 
+    def to_string(self) -> str:
```

Comment:
Unless I don't see something, you're using this method in a context of getting a unique item collection. However, the method does not use the entire state of the instance, so the logic for getting unique elements based on this method is broken. 

Assessing uniqueness is responsibility of `__hash__`. So, I recommend removing this method and relying on `__hash__` instead, if unique elements is what you're after.